jdbc/src/main/java/com/interface21/transaction/support/TransactionSynchronizationManager.java

```diff
@@ -1,23 +1,41 @@
 package com.interface21.transaction.support;
 
-import javax.sql.DataSource;
 import java.sql.Connection;
+import java.util.HashMap;
 import java.util.Map;
+import javax.sql.DataSource;
 
 public abstract class TransactionSynchronizationManager {
 
-    private static final ThreadLocal<Map<DataSource, Connection>> resources = new ThreadLocal<>();
+    private static final ThreadLocal<Map<DataSource, Connection>> resources = ThreadLocal.withInitial(HashMap::new);
 
-    private TransactionSynchronizationManager() {}
+    private TransactionSynchronizationManager() {
+    }
 
     public static Connection getResource(DataSource key) {
-        return null;
+        Map<DataSource, Connection> map = resources.get();
+        return map.get(key);
     }
 
     public static void bindResource(DataSource key, Connection value) {
+        Map<DataSource, Connection> map = resources.get();
+        if (map.containsKey(key)) {
+            throw new IllegalStateException("이미 커넥션이 존재합니다.");
+        }
+        map.put(key, value);
     }
 
     public static Connection unbindResource(DataSource key) {
-        return null;
+        if (!isBound(key)) {
+            throw new IllegalStateException("기존에 바인딩 된 연결이 존재하지 않습니다.");
+        }
+        Map<DataSource, Connection> map = resources.get();
+        Connection removed = map.remove(key);
+        return removed;
```

Comment:
인텔리제이에서 경고를 발생시키고 있네요.

<img width="361" height="80" alt="image" src="https://github.com/user-attachments/assets/f06d13ab-d552-4169-8c39-a245b51bf559" />

요렇게도 가능할 것 같습니다!
```suggestion
        return map.remove(key);
```

Reply:
remove 직후에 ThreadLocal 을 제거하는 로직이 추가되어 인라인 처리를 안 해도 될 것 같아요!